src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -140,19 +144,34 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
                     "Caught exception while establishing connection to AWS Iot");
 
             IotIdentityHelper iotIdentityHelper = iotIdentityHelperFactory.getInstance(connection);
-            CreateKeysAndCertificateResponse response;
-            response = FutureExceptionHandler.getFutureAfterCompletion(iotIdentityHelper.createKeysAndCertificate(),
-                    "Caught exception during PublishCreateKeysAndCertificate");
 
-            writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
+            String certificateOwnershipToken = null;
+            if (csrPath == null) {
+                CreateKeysAndCertificateResponse response;
+                response = FutureExceptionHandler.getFutureAfterCompletion(iotIdentityHelper.createKeysAndCertificate(),
+                        "Caught exception during PublishCreateKeysAndCertificate");
+
+                writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
+                certificateOwnershipToken = response.certificateOwnershipToken;
+            } else {
+                CreateCertificateFromCsrResponse response;
+                Path csrFile = Paths.get(csrPath);
```

Comment:
Probably want to add a `Files.exists` check and then produce a good error message if the file doesn't exist.

Reply:
Now there is a file check here. If the file doesn't exist, an InterruptedException is thrown.

Reply:
Interrupted shouldn't be used for this case. Where is that thrown? I don't see that anywhere here.

Reply:
Thank you, it should be thrown as an IOException such as `throw new IOException("CSR file does not exist")`. InterruptedException has a very particular meaning in Java which is why I requested that you not use that exception type.

Reply:
Please ignore this message. It seems it was saved before and sent now. I have added two new comments below about removing the file exist check and throwing a runtime error. I am marking this as resolved to prevent confusion.